ravenframework/Models/ROM.py

```diff
@@ -326,64 +328,97 @@ def train(self,trainingSet):
       @ In, trainingSet, dict or PointSet or HistorySet, data used to train the ROM; if an HistorySet is provided the a list of ROM is created in order to create a temporal-ROM
       @ Out, None
     """
-    if type(trainingSet).__name__ == 'ROM':
-      self.trainingSet              = copy.copy(trainingSet.trainingSet)
-      self.amITrained               = copy.deepcopy(trainingSet.amITrained)
-      self.supervisedContainer      = copy.deepcopy(trainingSet.supervisedContainer)
-      self.seed = trainingSet.seed
+    if isinstance(trainingSet, ROM):
+      self.trainFromInstance(trainingSet)
     else:
       # TODO: The following check may need to be moved to Dummy Class -- wangc 7/30/2018
-      if type(trainingSet).__name__ != 'dict' and trainingSet.type == 'HistorySet':
+      if type(trainingSet) != dict and trainingSet.type == 'HistorySet':
```

Comment:
First, could you add a description to list all possible data structures for trainingSet?
Second, could you add checks for different data structures for trainingSet?

Reply:
This looks like a specific check for history set alignment, right? I don't know if we need to find out all the different approaches to ROMs within this PR, do we? This sounds like a bigger issue.